packages/smart-wallet/src/utils.js

```diff
@@ -113,6 +112,27 @@ export const assertHasData = async follower => {
   }
 };
 
+/**
+ * Sequence currents from a wallet UpdateRecord publication feed. Note that local
+ * state may not reflect the wallet's state if the initial currents are missed.
+ *
+ * If this proves to be a problem we can add an option to this or a related
+ * utility to reset state from RPC.
+ *
+ * @param {ERef<Subscriber<import('./smartWallet').CurrentWalletRecord>>} currents
+ */
+export const sequenceCurrents = currents => {
+  const sequence = [];
+
+  void observeIteration(subscribeEach(currents), {
+    updateState: updateRecord => {
+      sequence.push(updateRecord);
+    },
+  });
+
+  return sequence;
```

Comment:
It's by design that this returns an empty array that in later turns gets update records pushed onto it? That seems worth a comment.

Reply:
I'd be much more comfortable if this were in a test-supports file rather than a `src/` file.

Reply:
me too. moved to inter-protocl tests supports